UM/PluginError.py

```diff
@@ -7,29 +7,18 @@
 
 ##  A general class for any error raised by a plugin.
 class PluginError(Exception):
-    def __init__(self, error = None): #pylint: disable=bad-whitespace
-        super().__init__()
-        self._error = error
-
-    def __str__(self):
-        return self._error
+    pass
 
 
 ##  Raised when a plugin could not be found.
-class PluginNotFoundError(Exception):
-    def __init__(self, name):
-        super().__init__()
-        self._name = name
-
+class PluginNotFoundError(PluginError):
     def __str__(self):
-        return "Could not find plugin " + self._name
+        name = self.args[0]     # pylint: disable=unsubscriptable-object
```

Comment:
I think you can just use cast (from typing import cast)  to resolve the typing error (eg; cast(str, self.args[0]))

Reply:
Using cast(str, self.args[0])

We still get the warning

```
Value 'self.args' is unsubscriptablepylint(unsubscriptable-object)
```
from pylint 

Please refer to update to the pull request for proposed alternative solution

Reply:
Ah, I stand corrected then.